[F10-2] LoanBook

wn opened the pull request:
PR to get notifications from tutor. @prokarius @FongYuan @Kelly9373 @xantho09

wn committed:
Set name and url of gradle build to reflect team

wn committed:
Merge pull request #1 from CS2103-AY1819S1-F10-2/set-github-url-and-site-name

Set name and url of gradle build to reflect team

wn committed:
Update badges

wn committed:
Add weineng's profile

wn committed:
Merge pull request #3 from CS2103-AY1819S1-F10-2/weineng-profile

Add weineng's profile

wn committed:
Merge branch 'master' into update-badge

wn committed:
Merge pull request #2 from CS2103-AY1819S1-F10-2/update-badge

Update badges

Kelly9373 committed:
Update Xiaohang and Fong Yuan part in AboutUs and add portfolio

Kelly9373 committed:
Update the UserGuide (#5)

* Update AboutUs and add my photo and github link

* Add my portfolio

* Link to my portfolio

* Update Xiaohang's part in AboutUs

* Solve an error in AboutUs line 27

* Try to solve the trailing white space error

* Update Fong Yuan's part in AboutUs and add his portfolio

* Update Fong Yuan's portfolio

* Update introduction and quick start in UserGuide

* Update the feature part in UserGuide

* Solve the trailing whitespace error on line 160 and line 162

* Change toc and toc topic in UserGuide

* Fix toc in UserGuide

wn committed:
Update contact details and weineng profile

Commit:
Correct the AboutUs photos' name

Commit:
Correct the photos' name in AboutUs

Commit:
Force name change for FongYuan.jpg

Commit:
Correct the names from .jpg to .png

Commit:
Correct the errors in AboutUs

Kelly9373 committed:
Correct the name of the photos in AboutUs (#13)

* Correct the AboutUs photos' name

* Correct the photos' name in AboutUs

* Force name change for FongYuan.jpg

* Correct the names from .jpg to .png

* Correct the errors in AboutUs

prokarius committed:
Add lines to gitignore; Add prokarius picture

prokarius committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-F10-2/main into david-picture

Commit:
Update roles and responsibilities of Wei Neng, Xiaohang and Fong Yuan in AboutUs

Commit:
Change the title and a few info in DevGuide

xantho09 committed:
Add Irham's picture (#15)

Kelly9373 committed:
Merge branch 'master' into AboutUs

Kelly9373 committed:
Merge branch 'master' into DevGuide

prokarius committed:
Fix merge conflict in AboutUs.adoc

OrangeJuice7 committed:
Merge pull request #18 from Kelly9373/DevGuide

Change the title author team and date in DevGuide

OrangeJuice7 committed:
Merge branch 'master' into AboutUs

OrangeJuice7 committed:
Merge branch 'master' into david-picture

OrangeJuice7 committed:
Merge branch 'master' into update-docs

OrangeJuice7 committed:
Merge pull request #19 from CS2103-AY1819S1-F10-2/david-picture

Add David's picture

Commit:
Merge branch 'AboutUs'

Commit:
Merge branch 'AboutUs'

Kelly9373 committed:
Merge branch 'master' into master

OrangeJuice7 committed:
Merge pull request #20 from Kelly9373/master

Update roles and responsibilities of Wei Neng, Xiaohang and Fong Yuan in AboutUs

OrangeJuice7 committed:
Merge branch 'master' into update-docs

OrangeJuice7 committed:
Merge pull request #12 from CS2103-AY1819S1-F10-2/update-docs

Update contact details

Kelly9373 committed:
Update README, including changing the app's name and its introduction (#21)

OrangeJuice7 committed:
Delete obsolete profile images

OrangeJuice7 committed:
Update Ui.png (#26)

wn committed:
Remove unused text

wn committed:
Merge branch 'master' into removePics

parvathy-sudhir-pillai reviewed (commented):
Good effort, team! 

Few suggestions below:
* Introduction in User Guide does not refer to your LoanBook app?
* Update Appendices iN Developer Guide with Product Scope, User stories, Usecases, NFR and Glossary.

Will do a full review after v1.1 as your new tutor.

OrangeJuice7 committed:
Merge pull request #25 from FongYuan/removePics

Delete obsolete profile images

OrangeJuice7 committed:
Merge branch 'master' into update-about-us

OrangeJuice7 committed:
Merge pull request #28 from CS2103-AY1819S1-F10-2/update-about-us

Remove obsolete text in AboutUs.adoc

wn committed:
DeveloperGuide: update "Setting up" section (#23)

Due to an ongoing issue[1] with some of the newer versions of IntelliJ,
the IDE may report code errors for JAXB or JavaFx APIs depending on the
Intellij Edition being used, although the code can be build and run
successfully since we have already imported the dependencies through
gradle.

This may cause confusion amongst users.

Let's update the "Setting up" section to address this issue, and how
to resolve it.

[1]: https://youtrack.jetbrains.com/issue/IDEA-189060

wn committed:
Add codacy and MIT licence badge (#32)

OrangeJuice7 committed:
Rename Person class to Loan. (#10)

* rename person package to loan

* Rename Person class to Loan, as well as all related classes, methods and variables.

* Rename XmlAdaptedPerson to XmlAdaptedLoan.

* Rename PersonBuilder to LoanBuilder.

* Rename some misc person classes, methods and variables into loans.

* Rename all XML files that have 'Person' in their name to the corresponding 'Loan' name.

* Rename all instances of <person> or <persons> in the XML files with <loan> and <loans> respectively.

* Thorough sweep of everything that still mentions 'person' in the project. Only instances left are the personal particulars in UserGuide.adoc and HelpWindow.html.

* Swap 'import [...].loan.Loan' with '[..].loan.Name' to maintain lexicographical order.

* More checkstyle corrections.

* More checkstyle corrections for DeleteCommandSystemTest.java.

* Correct typos in the license files

* Correct capitalization typo in typicalLoansAddressBook.xml

* Correct import scope issue in LoanUtil.java

* Correct indentation in FindCommandSystemTest.java

parvathy-sudhir-pillai reviewed (commented):
Overall:
Good start, Team. However, some areas need significantly more work. Hope you can tackle those areas in upcoming features.

* Readme adoc
1. Suggest to move the reference to AB-4 to the Acknowledgements section from the Product Description

* User Guide adoc
1. Introduction section refers to WFIDX and not LoanBook. Revise this.
2. Suggest adding a glossary for the abbreviations used in the User Guide. It needs to cater to novices also
3. Use formal language in your User Guide (last point in NFR)

* Developer guide adoc
1. Update Appendix B: Product Scope
2. Add more user stories and use cases. These are just too few and many important ones are missing e.g. use case for the summary command.

xantho09 committed:
Change title to "Loanbook App" (#30)

* Add Irham's picture

* Change title to Loanbook App

OrangeJuice7 committed:
Update Dev Guide section 1.2 (Setting up in com) (#33)

Corrected mistakes and made instructions more clear in section 1.2 "Setting up the project in your computer" of the Dev Guide.

OrangeJuice7 committed:
Codacy check (#34)

* Change equals() to == for null object test

* Remove unused private methods in LoanListPanel.java

* Extract methods to reduce NPath complexity in XmlAdaptedLoan.java

* Remove underscores in test names in XmlAdaptedLoanTest.java

* Remove underscores in test names in UniqueLoanListTest.java

* Remove unused imports in LoanListPanel.java

* Add missing javadoc comments in XmlAdaptedLoan.java

* Revert "Remove unused private methods in LoanListPanel.java"

This reverts commit 6f2538697ec5868c526883c6cabfa75ffdbe9ad1.

Restore 'unused private methods' in LoanListPanel.java; actually used
by the event bus thru @Subscribe

* Add missing javadoc comments in XmlAdaptedLoan.java

* Revert "Remove unused imports in LoanListPanel.java"

This reverts commit 1126d93e4eb460656ec70da47a0a0b37175af86c.

wn committed:
Update introduction of user guide

OrangeJuice7 committed:
Update Readme (#38)

- Moved AB4 reference to the Acknowledgements section, as suggested by Ms Parvathy
- Changed target user group to bicycle rental shop owners

wn committed:
Fix grammatical issues

OrangeJuice7 committed:
Merge pull request #39 from CS2103-AY1819S1-F10-2/update-user-guide

Update introduction of user guide

prokarius committed:
Add use cases in dev guide (#44)

OrangeJuice7 committed:
Add user stories

OrangeJuice7 committed:
Update product scope in Dev Guide

OrangeJuice7 committed:
Remove duplicate user story

OrangeJuice7 committed:
Remove another duplicate user story

OrangeJuice7 committed:
Merge pull request #45 from FongYuan/DevGuideUserStories

Added user stories in Dev Guide

OrangeJuice7 committed:
Add DataField class (#35)

* Add DataField.java

* Update Name class to inherit from DataField

* Rename variable Name.fullName to Name.value

* Update Phone class to inherit from DataField

* Correct indentation in Phone.java

* Update Email classs to inherit from DataField

* Update Address class to inherit from DataField

* Update Tag class to inherit from DataField

* Rename Tag.tagName to Tag.value

* Update Model class diagram in docs

* Correct comments and checkstyle errors in DataField.java

* Correct indentations in Phone.java and Email.java

prokarius committed:
Add LoanTime class (#24)

* Add LoanTime class

* Fix to pass styleguide

* Fill up LoanTime input string constructor

* Add date and time validations checks in LoanTime class

* Add validation tests for LoanTime class

* Fix regex on time parsing

* Add tests and fix code for loanTimeDifference method

* Fix to pass stylecheck

* Fix LoanTime class based on comments

* Fix and add more LoanTime testcases

* Add more input validation to isValidTime method

xantho09 committed:
Refactor LoanTime to extend DataField class (#73)

* Add Irham's picture

* Change title to Loanbook App

* Refactor LoanTime to extend DataField class

* Edit documentation of LoanTime methods

* Remove unnecessary import statement from LoanTimeTest

* Fix style check violations in LoanTime and LoanTimeTest

* Fix test in LoanTimeTest

* Modify loanTimeDifferenceMinutes method in LoanTime to use static method

Kelly9373 committed:
Add LoanRate class (#29)

* Update README, including changing the app's name and its introduction

* Add LoanRate class

* Correct the checkstyle error

* Correct checkstyle errors in LoanRateTest

* Add a new line in LoanRateTest

* Add a new line in LoanRate class

* Update LoanRate class

* Change the regex of rate and add some test cases

* Change the type of rate from String to double

* Correct checkstyle errors

* Update LoanRate class

* Correct checkstyle error

* Correct checkstyle errors

OrangeJuice7 committed:
Update DataField documentation

OrangeJuice7 committed:
Remove Email validity predicate

OrangeJuice7 committed:
Remove Address validity predicate

OrangeJuice7 committed:
Remove Name validity predicate

OrangeJuice7 committed:
Remove Phone validity predicate

OrangeJuice7 committed:
Remove unused import statements

OrangeJuice7 committed:
Remove LoanTime validity predicate

OrangeJuice7 committed:
Make LoanRate inherit from DataField

Commit:
Add config.json

xantho09 committed:
Modify LoanTime toString method to match constructor input (#79)

* Add Irham's picture

* Change title to Loanbook App

* Refactor LoanTime to extend DataField class

* Edit documentation of LoanTime methods

* Remove unnecessary import statement from LoanTimeTest

* Fix style check violations in LoanTime and LoanTimeTest

* Fix test in LoanTimeTest

* Modify loanTimeDifferenceMinutes method in LoanTime to use static method

* Modify toString method of LoanTime to match constructor input

* Rename parseLoanTime method in LoanTime to parseInstant

* Add additional test for LoanTime toString method

* Resolve merge conflict

* Remove magic strings in LoanTimeTest

* Fix constructorInputCheckCurrentDate method in LoanTimeTest

OrangeJuice7 committed:
Merge pull request #87 from Kelly9373/add-config.json

Add config.json for RepoSense

OrangeJuice7 committed:
Merge branch 'master' into DataFieldPred

OrangeJuice7 committed:
Add Bike class (#82)

https://github.com/CS2103-AY1819S1-F10-2/main/issues/70

It is still an orphaned class for now; no classes or commands are associated with it.

OrangeJuice7 committed:
Add Bike Exceptions

OrangeJuice7 committed:
Add UniqueBikeList class

OrangeJuice7 committed:
Update AddressBook and ReadOnlyAddressBook

OrangeJuice7 committed:
Update AddressBookTest to follow new ROAB interface

OrangeJuice7 committed:
Update Model and ModelManager

OrangeJuice7 committed:
Update ModelStub class in AddCommandTest to follow new Model interface

OrangeJuice7 committed:
Update Model class diagram

OrangeJuice7 committed:
Add newline char at EOF in DuplicateBikeException.java

OrangeJuice7 committed:
Correct import order for AddressBook.java and UniqueBikeList.java

OrangeJuice7 committed:
Update tests for AddressBook

OrangeJuice7 committed:
Update tests for ModelManager

OrangeJuice7 committed:
Document Bicycle Management in Dev Guide

OrangeJuice7 committed:
Add DeleteLoanSequenceDiagram.png

OrangeJuice7 committed:
Add note for using two lists of Loans

wn committed:
Add nric class (#36)

Fixes issue #43

OrangeJuice7 committed:
Move test Bikes to TypicalBikes.java

OrangeJuice7 committed:
Add tests for UniqueBikeList

OrangeJuice7 committed:
Update AddressBookBuilder

OrangeJuice7 committed:
Update ModelManager tests

OrangeJuice7 committed:
Update AddressBook tests

OrangeJuice7 committed:
Merge branch 'master' into BikeToModel

OrangeJuice7 committed:
Correct checkstyle errors

OrangeJuice7 committed:
Merge github.com:CS2103-AY1819S1-F10-2/main into BikeToModel

Update from team repo

OrangeJuice7 committed:
Merge branch 'BikeToModel' of github.com:FongYuan/main into BikeToModel

Update from remote repo

wn committed:
Add new colour to tags (#90)

Following https://github.com/se-edu/addressbook-level4/pull/798/files#diff-16e8193ce6874fb434b8aa49d25422f0

xantho09 committed:
Add ./config.json to .gitignore file (#97)

The config.json file created by the AddressBook should be ignored by git.

wn committed:
Add password for deletion (#96)

Fixes first point of https://github.com/CS2103-AY1819S1-F10-2/main/issues/40

Instead of confirming, require password.

prokarius committed:
Add return feature documentation in devguide (#94)

* Add return feature documentation in devguide

* Remove trailing whitespaces

* Edit based on FongYuan's comments

* Edit Devguide to reflect change in spec

OrangeJuice7 committed:
Fix nits

xantho09 committed:
Fix .gitignore to correctly ignore config.json (#98)

* Add Irham's picture

* Change title to Loanbook App

* Refactor LoanTime to extend DataField class

* Edit documentation of LoanTime methods

* Remove unnecessary import statement from LoanTimeTest

* Fix style check violations in LoanTime and LoanTimeTest

* Fix test in LoanTimeTest

* Modify loanTimeDifferenceMinutes method in LoanTime to use static method

* Modify toString method of LoanTime to match constructor input

* Rename parseLoanTime method in LoanTime to parseInstant

* Add additional test for LoanTime toString method

* Resolve merge conflict

* Remove magic strings in LoanTimeTest

* Fix constructorInputCheckCurrentDate method in LoanTimeTest

* Add ./config.json to .gitignore

* Fix .gitignore to correctly ignore config.json

xantho09 committed:
Create Loan ID data field (#100)

* Add Irham's picture

* Change title to Loanbook App

* Refactor LoanTime to extend DataField class

* Edit documentation of LoanTime methods

* Remove unnecessary import statement from LoanTimeTest

* Fix style check violations in LoanTime and LoanTimeTest

* Fix test in LoanTimeTest

* Modify loanTimeDifferenceMinutes method in LoanTime to use static method

* Modify toString method of LoanTime to match constructor input

* Rename parseLoanTime method in LoanTime to parseInstant

* Add additional test for LoanTime toString method

* Resolve merge conflict

* Remove magic strings in LoanTimeTest

* Fix constructorInputCheckCurrentDate method in LoanTimeTest

* Create LoanId class

* Create tests for LoanIdTest

* Add logic for largest possible LoanId

* Add test for Maximum Loan ID

* Add documentation to LoanId class

* Fix checkstyle issues

* Fix a comment in LoanIdTest

OrangeJuice7 committed:
Fix DeleteCommand usage message (#102)

wn committed:
Fix gitignore

OrangeJuice7 committed:
Correct style

Kelly9373 committed:
Add hide private info documentation in dev guide (#106)

* Add hide private info documentation in dev guide

* Add more content

* Update pptx and png

* Do a little adjustment to the LoanListPanelView.pptx and png

* Fix errors in HideInfoSequenceDiagrams.pptx and png

* Fix some grammatical and language style errors

OrangeJuice7 committed:
Merge branch 'master' into IntegrateLoanRate

OrangeJuice7 committed:
Merge pull request #81 from FongYuan/IntegrateLoanRate

Make LoanRate inherit from DataField

OrangeJuice7 committed:
Merge branch 'master' into DataFieldPred

wn committed:
Merge branch 'master' into BikeToModel

OrangeJuice7 committed:
Merge pull request #92 from FongYuan/BikeToModel

Add Bikes to AddressBook and Model

OrangeJuice7 committed:
Merge branch 'master' into DataFieldPred

OrangeJuice7 committed:
Merge pull request #80 from FongYuan/DataFieldPred

Change DataField Predicate requirement

OrangeJuice7 committed:
Merge branch 'master' into DevGuideEnhance

OrangeJuice7 committed:
Add XmlAdaptedBike.java

OrangeJuice7 committed:
Add tests for XmlAdaptedBike

OrangeJuice7 committed:
Update XmlSerializableAddressBook with bikes

Kelly9373 committed:
Refactor address book to loan book (#115)

* Update README, including changing the app's name and its introduction

* Add LoanRate class

* Correct the checkstyle error

* Correct checkstyle errors in LoanRateTest

* Add a new line in LoanRateTest

* Add a new line in LoanRate class

* Update LoanRate class

* Change the regex of rate and add some test cases

* Change the type of rate from String to double

* Correct checkstyle errors

* Update LoanRate class

* Correct checkstyle error

* Correct checkstyle errors

* Add config.json

* . (#2)

* Refactor LoanTime to extend DataField class (#73)

* Add Irham's picture

* Change title to Loanbook App

* Refactor LoanTime to extend DataField class

* Edit documentation of LoanTime methods

* Remove unnecessary import statement from LoanTimeTest

* Fix style check violations in LoanTime and LoanTimeTest

* Fix test in LoanTimeTest

* Modify loanTimeDifferenceMinutes method in LoanTime to use static method

* Add LoanRate class (#29)

* Update README, including changing the app's name and its introduction

* Add LoanRate class

* Correct the checkstyle error

* Correct checkstyle errors in LoanRateTest

* Add a new line in LoanRateTest

* Add a new line in LoanRate class

* Update LoanRate class

* Change the regex of rate and add some test cases

* Change the type of rate from String to double

* Correct checkstyle errors

* Update LoanRate class

* Correct checkstyle error

* Correct checkstyle errors

* Add a new line at EOF

* Replace "address" by "loan" that won't affect other parts of the project in logic

* Replace "address" by "loan" that won't affect other parts of the project in common

* Replace "address" by "loan" that won't affect other parts of the project in model

* Replace "address" by "loan" that won't affect other parts of the project in storage

* Replace "address" by "loan" that won't affect other parts of the project in MainApp

* Refactor AddressBook class to LoanBook

* Refactor VersionedAddressBook class to VersionedLoanBook

* Refactor ReadOnlyAddressBook class to ReadOnlyLoanBook

* Refactor "address" to "loan" in model folder and model test

* Replace all "address" to "loan" in logic folder and logic tests

* Replace "address" by "loan" in common and ui, but some parts involving storage are left

* Refactor all the files and tests about Storage

* Refactor "address" to "loan"

* Fix CheckStyle Errors

* Fix CheckStyle Error

* Fix a missing change

OrangeJuice7 committed:
Resolve merge conflicts in XmlSerializableLoanBook.java

OrangeJuice7 committed:
Merge branch 'FongYuan-XMLBikes' into XMLBikes

Merge banch with team repo

OrangeJuice7 committed:
Merge branch 'master' into DevGuideEnhance

OrangeJuice7 committed:
Merge pull request #95 from FongYuan/DevGuideEnhance

Document Bicycle Management in Dev Guide

OrangeJuice7 committed:
Remove newline in XmlAdaptedBikeTest

OrangeJuice7 committed:
Add SerializableLoanBook XML test files

OrangeJuice7 committed:
Update XmlSerializableLoanBook tests

Kelly9373 committed:
Merge branch 'master' into XMLBikes

OrangeJuice7 committed:
Merge pull request #112 from FongYuan/XMLBikes

Add XML support for bikes

OrangeJuice7 committed:
Rename images (#116)

For https://github.com/CS2103-AY1819S1-F10-2/main/issues/110

There are no associated .pptx files with the images.

parvathy-sudhir-pillai reviewed (commented):
### Feedback for v1.2

**User Guide**
* Few commands refer to the index, while others refer to load id. Are they the same? Use uniformly as interchanging may confuse the user.
![f10-1](https://user-images.githubusercontent.com/9066342/47268073-b7243880-d57e-11e8-94d2-7c4041e51e80.png)

**Developer Guide**
* Remove the second class diagram under Model component or match it to your class diagram.
* Can remove section 3.2  [Proposed] Data Encryption
* Include UML diagrams for Return a Loan and Bicycle Management features
* Can remove Appendix A and other unnecessary details from dev guide that refer to AB-4
* Limit each use case step to have a single operation and not a sequence of operations.
![f10-2](https://user-images.githubusercontent.com/9066342/47268231-dc19ab00-d580-11e8-9fc8-761b106d476f.png)
* Use the right quotes in punctuation
![f10-3](https://user-images.githubusercontent.com/9066342/47268263-4d595e00-d581-11e8-8a22-8d811797491a.png)
* Hide private information feature doesn't involve the Logic component? Better to include sequence diagrams with in component interactions.
![f10-4](https://user-images.githubusercontent.com/9066342/47268365-77f7e680-d582-11e8-851e-e91f3a2e5b25.png)
* Not enough new/updated UML diagrams to give feedback on. Please try to add more soon.

### Overall
* Good job on adding/updating feature implementation descriptions. 👍
* Draw sequence diagrams for the sequence of steps involved in the execution of command. Keep in mind, however, that we discourage low level diagrams that span multiple components. Try to draw diagrams at multiple levels: one that shows component level interactions and others shown low level interactions within one components.
* Catch up on the documentation of UML diagrams.
* Let's push the product towards a user-testable state by v1.3

OrangeJuice7 committed:
Extract TypicalLoanBook and add Bike support (#120)

* Extract TypicalLoanBook to its own file

* Update location of TypicalLoanBook for everyone

* Add typical bikes to TypicalLoanBook

* Add typical bikes to typicalLoansLoanBook.xml

* Rename typicalLoansLoanBook.xml to typicalLoanBook.xml

* Correct checkstyles

OrangeJuice7 committed:
Dev guide update (#119)

* Remove 'Better OOP Diagram' from Model section

* Update Model section AddressBook to LoanBook

* Remove Data Encryption section

* Remove Appendix A: Suggested Programming Tasks to Get Started

* Correct formatting errors in Bicycle Management section

* Add class diagram to Bicycle Management section

* Update use cases

* Rename AddressBook to LoanBook

* Rename AddressBook to LoanBook in DeleteLoanSequenceDiagram

* nits

* Change tab char to spaces at line 996

OrangeJuice7 committed:
Add addbike command (#111)

* Add AddBikeCommand.java

* Add AddBikeCommandParser.java

* Extract ModelStub class from AddCommandTest into its own file in testutil

* Add tests for AddBikeCommand

* Change TypicalBikes to use data from CommandTestUtil

* Add tests for AddBikeCommandParser

* Main command parser supports addbike command

* Add BikeUtil and AddressBookParser test

* Update new files with refactoring

* Fix Bike toString test

wn committed:
Set password command (#101)

Todo:

- [x] Add test
- [x] Replace all existing password stubs of `a12345` to use UserPref password.
- [ ] Update user guide 
- [ ] Update developer guide

Kelly9373 committed:
Add datafields into Loan class (#124)

* Add datafields into Loan class

* Correct checkstyle error

* Correct checkstyle error again

* Correct CheckStyle Errors

* Add more test cases in XmlAdaptedLoanTest class to increase coverage

* Remove unused import

prokarius committed:
Update dev and user guides (#126)

Added more content for guides in terms of summary command.

Fixed some minor English and augmented some details in the guides.

Partially Fix #117 as all index were changed to LOAN_ID

Kelly9373 committed:
Hide data(Censor sensitive information of customers) (#125)

* Add datafields into Loan class

* Correct checkstyle error

* Correct checkstyle error again

* Correct CheckStyle Errors

* Add more test cases in XmlAdaptedLoanTest class to increase coverage

* Remove unused import

* Hide private data in LoanCard and adjust the format of the LoanCard

* Add Censor Interface and let Email, Phone and Nric class implements it

* Show Censored info on LoanCard

* Update test cases

* Correct checkstyle errors

* Update Censor method

OrangeJuice7 committed:
Extract common functionality between UniqueLoanList and UniqueBikeList (#123)

* Add UniqueListItem.java

* Add UniqueList.java

* Make Bike class implement UniqueListItem

* Make Loan class implement UniqueListItem

* Make UniqueBikeList extend UniqueList

* Make UniqueLoanList inherit from UniqueList

* Update Model class diagram

* Remove trailing whitespace

* UniqueList rename set() to setAll() to prevent overloading

* Correct javadoc style

prokarius committed:
Add LoanStatus enum (#105)

* Add LoanStatus enum

* Add tests for LoanStatus enum

* Add tests for LoanStatus enum in LoanBuilder class

* Save LoanStatus to storage component

* Change XMLAddressBookStorage test data to fit new XML format

* Add Tests for LoanStuatus

* Fix nits

wn committed:
Add user confirmation to Developer guide (#104)

Partially fix issue https://github.com/CS2103-AY1819S1-F10-2/main/issues/93

Kelly9373 committed:
Update Hide Info's DG (#132)

* Update Hide Info's DG

* Make small changes

OrangeJuice7 committed:
Refactor XmlAdaptedLoan (#131)

* Add checkFieldValid()

* Update toModelType()

* Remove old private methods

Kelly9373 committed:
Update Devguide's images (#134)

* Update diagrams in DG

* Update images

* Update remain images

prokarius committed:
Fix Loan class data fields (#133)

* Refactor LoanTime time->startTime & add endTime

* Set add function to take current time as startTime

* Merge branch 'master' into update-loanfield

* Fix previous incorrectly made testcases

* Fix inconsistent test cases

* Fix failing testcases

* Fix system tests

* Update LoanCards Javafx

* Fix nits

xantho09 committed:
Create Loan ID Manager class (#113)

Fix issue #99.

OrangeJuice7 committed:
Check existing bikes when adding a loan (#135)

* Add getBike() in UniqueBikeList

* Add getBike() to LoanBook

* Remove non-static initialization block in LoanBook.java

* Remove ModelStub from SetPasswordCommandTest.java

* Remove ModelStub from AddCommandTest.java

* Fix SetPasswordCommandTest.java

* Add getBike() to Model

* Add getBike check in AddCommand and update tests

* Update AddBikeCommand messages

* Fix ClearCommand

* Rename resetData to replaceData in Model

wn committed:
Update user guide (#137)

OrangeJuice7 committed:
Rename seedu.address package to loanbook (#138)

wn committed:
Rename addressbook to loanbook (#140)

wn committed:
Fix Nric parser (#144)

prokarius committed:
Add updates to guides (#142)

* Add updates to guides

wn committed:
Remove address (#146)

* Remove address field

Kelly9373 committed:
Update UGDG (#149)

* Update UGDG

* Correct language style in Dev Guide

* Correct language style in User Guide

* Update guides

* Update guides

* Fix typo

* Fix language nits in Dev Guide

OrangeJuice7 committed:
listbikes command (#150)

* Add ListCard class

* Make LoanCard inherit from ListCard

* Add BikeCard class and fxml

* Add ListPanel.java

* Make LoanListPanel inherit from ListPanel

* Update MainWindow

* Add BikeListPanel

* Add ListBikes command

* Implement BikeListShowEvent

* Implement LoanListShowEvent

* Add listbikes command test

* Add listbikes test in LoanBookParserTest.java

* Add BikeCard tests

* Add BikeListPanel tests

* Add ListCardHandle and ListPanelHandle

* Make BikeCardHandle and LoanCardHandle inherit from ListCardHandle

* Make BikeListPanelHandle and LoanListPanelHandle inherit from ListPanelHandle

OrangeJuice7 committed:
Update component class diagrams (#152)

* Update Model class diagram

* Update Storage class diagram

* Update UI class diagram

OrangeJuice7 committed:
Change Name to accept ' - , . characters (#153)

* Edit name regex and message_contraints

* Add extra tests for Name

wn committed:
Update guides (#148)

* Add updates to guides

prokarius committed:
Add return functionality (#159)

Add functionality to return a loan.

Wrote tests for said functionality and auxiliary functions as well.

Addresses issue #59

OrangeJuice7 committed:
Add colours to loan and bike lists (#157)

The list of loans in the UI is now yellow, and the list of bikes (accessible via `listbikes`) is blue.
This has been done to differentiate the list of loans from the list of bikes, so the user can more easily notice which list they are looking at.

Kelly9373 committed:
Add send email feature (#151)

* Update UGDG

* Correct language style in Dev Guide

* Correct language style in User Guide

* Update guides

* Update guides

* Fix typo

* Fix language nits in Dev Guide

* Write tests

* Add send reminder email feature.

* Change import sequence

* Fix import sequence

* Fix check style errors

* Fix checkstyle errors

* Update the feature

* Update gitignore

* Update messages and exceptions in remind command

* Improve it

* Fix checkstyle errors

* Update this feature

* Fix checkstyle error

* Fix checkstyle error and improve SendReminder class

* improve it

wn committed:
Fix sequence diagram (#166)

OrangeJuice7 committed:
Fix edit command not working with bikes (#156)

* Fix edit command not working with bikes

* Add tests for add and edit commands

* Remove LoanTimes from EditCommand

OrangeJuice7 committed:
Update docs (#165)

* Update user guide

* Fix Model class diagram

* Rewrite Model section in dev guide

* Update Bicycle Management class diagram

* Update Product Scope in Dev Guide

* Remove duplicate user story in dev guide

* Add use case for addbike command in dev guide

* Add manual testing docs in dev guide

xantho09 committed:
Integrate Loan ID into Loan class (#162)

* Integrate Loan ID into Loan class

* Add unique Loan ID and legal Loan ID Manager restriction checks

* Fix checkstyle errors

* Fix checkstyle issues

* Fix checkstyle issues

OrangeJuice7 committed:
Update logo (#167)

For https://github.com/CS2103-AY1819S1-F10-2/main/issues/139

wn committed:
Add search feature (#158)

Resolve https://github.com/CS2103-AY1819S1-F10-2/main/issues/55

TODO:

- [x] Check valid Dates (Write date class? Call library?)
- [x] Check if ENDDATE < STARTDATE
- [x] Write tests

xantho09 committed:
Add Loan ID details to Developer Guide (#168)

* Add Loan ID details to Developer Guide

wn committed:
Update app version (#169)

wn committed:
Update user guide and write up draft for PPP (#122)

* Write up draft for PPP

* Update intro

* Update PPP

* Add search feature to user guide

Kelly9373 committed:
Update UG (#172)

wn committed:
Hot fix (#176)

OrangeJuice7 committed:
Make add, addbike and delete commands inherit from ArgumentParser (#155)

* Make add, addbike and delete commands inherit from ArgumentParser

prokarius committed:
Fix grammar and nits UG (#245)

* Fix grammar issues #213, #214

* Remove non implemented non-function features section from UG

* Capitalise Gmail. #207

* Fix grammar issue #199

* Remove unnecessary space #200

* Fix more grammar issues #194 #192 #189 #188

* Fix unnecesary space #181

* Fix punctuation #180 #179

* Remove spaces before colons

wn committed:
Update app version (#175)

OrangeJuice7 committed:
add deletebike command (#178)

* Add DeleteBikeCommand

* Add DeleteBikeCommandParser

* Add DeleteCommand parser tests

* Add DeleteBikeCommand tests

OrangeJuice7 committed:
Add editbike command (#230)

* Add EditBike command

* Add editbike command parser

* Add EditBikeDescriptor builder

* Add EditBikeDescriptor tests

* Add editbike command parser tests

* Add editbike command tests

prokarius committed:
Fix issues in UG examples (#246)

* Fix grammar issues #213, #214

* Remove non implemented non-function features section from UG

* Capitalise Gmail. #207

* Fix grammar issue #199

* Remove unnecessary space #200

* Fix more grammar issues #194 #192 #189 #188

* Fix unnecesary space #181

* Fix punctuation #180 #179

* Remove spaces before colons

* Fix SVA for #198

* Change examples to work when testers are copy pasting #232

* Fix add command examples #219 #216 #186

* Fix edit command examples #191 #217

* Fix incorrectly formatted tip in add documentation #187

OrangeJuice7 committed:
Fix equals(null) tests (#247)

Kelly9373 committed:
fix UG errors (#248)

prokarius committed:
Add summary functionality (#251)

* Add summary command

* Add summary command tests

* Fix stylecheck

* Shift Summary class to its own file

prokarius committed:
Increase coverage across all files (#252)

* Add tests for storage component

* Add tests for model component

* Write tests for setpassword command parser

* Fix checkstyle

xantho09 committed:
Add getLoanById method and LoanID parser (#254)

* Add getLoanById method in Model

* Add tests for getLoanById

* Add Loan ID prefix and parser

* Add tests for Loan ID Parsing

* Fix checkstyle issues

OrangeJuice7 committed:
Move ResultDisplay to BrowserPanel (#177)

wn committed:
Fix issue from dry run (#255)

* Fix issue #208

* Add future release table

* Fix NRIC error message, issue #201

* Fix user guide

* Fix user guide

* Fix lint

* Update docs

* Update NRIC error message

* Fix lint

xantho09 committed:
Add ResetAll command and rename Clear to ResetLoans (#256)

* Fix resetLoans to only reset loans and not reset ID

* Add resetBikes method in Model

* Add ResetAll command

* Fix documentation of ResetAllCommand

* Rename Clear command to Reset command

* Add tests for ResetAllCommand and parser

* Fix checkstyle errors

* Rename Reset command to ResetLoans

* Create PasswordProtectedCommand class

* Refactor ResetAll and Delete commands to extend PasswordProtectedCommand

* Add password authentication to ResetLoansCommand

* Modify reset commands in User Docs

* Fix checkstyle issues

Kelly9373 committed:
Fix remind command bugs and change its command format (#253)

* Fix remind command bugs and change its command format

* Fix checkstyle errors

* update it

* Fix checkstyle errors

* Fix it

wn committed:
Add status to tag (#163)

* Add status to tag

* Add files

* Remove tests for tag equality in LoanCard

* Remove stray codes

prokarius committed:
Change return time display from PLACEHOLDER to actual (#261)

Kelly9373 committed:
Fix setemail bugs and change the command format of setemail (#249)

* Fix setemail bugs and change the command format of setemail

* Fix tests

* Fix checkstyle error

* Fix it

* Fix it

* Fix it

* Fix checkstyle error

wn committed:
Update Ui.png (#266)

wn committed:
Hide pass in history (#262)

* Fix editbike command not checking for dupe bikes

* Hide password

* Fix lint

* Fix test

* Revert "Fix editbike command not checking for dupe bikes"

This reverts commit 18c93821a42d5aeee100a1efdf069a15574b57ed.

wn committed:
Update PPP (#265)

* Update PPP

* Add PR

* Provide credit for code

* Update user guide

* Update DG

* Improve dg

* Add image

OrangeJuice7 committed:
Fix editbike command not checking for dupe bikes (#258)

parvathy-sudhir-pillai reviewed (commented):
### Mid v1.4 review
* Dev guide: Update the "undo" feature SD with multi-level SD's. CS2103 encourages multi-level sequence diagrams. Higher level interactions between components (e.g. Logic and Model) in a separate diagram and lower level interactions between sub-components (e.g. Parser and Command) within a component (Logic) in a separate diagram. The lower level diagram should not include interactions between components (e.g. Logic and Model).
* Add behaviour (Sequence or Activity) diagram for "Return a Loan" and set password feature/command.

**Good job, Team!
All the best for your v1.4 submission and presentation!**

OrangeJuice7 committed:
Update Messages (#260)

* Categorize messages

* Update unknown command message

* Update invalid command format message

* Update invalid index message

* Move Duplicate Bike message from addbike command to common messages

* Move bike not found message to common messages

* Update loan verification messages

* Move not edited message to common messages

* Update unknown command message

* Update messages

* Rename MESSAGE_DUPLICATE_FAILURE to MESSAGE_SAME_AS_OLDEMAIL

* Rename MESSAGE_INVALID_INFO to MESSAGE_INVALID_LOAN

* Update SetEmailCommandTest.java

wn committed:
Hot fix for image name (#268)

* Hot fix for image name

* Update image

wn committed:
Encrypt password (#264)

* Encrypt password

* Provide credit for code

* Provide credit for code

* Fix merge conflict

* Add abstract class to commands

* Fix test

wn committed:
Update history command: (#267)

* Update history command:

* Fix english

wn committed:
Hot fix image name

prokarius committed:
Update Documentations (#263)

* Add list of params to functions in UG

* Add UML Diagrams to Docs

* Fix based on FY's comments

* Fix minor nits

prokarius committed:
Upload David PPP (#269)

* Upload David PPP

* Remove trailing whitespace

prokarius committed:
Update PPP (#271)

* Upload David PPP

* Remove trailing whitespace

* Fix PPP

Kelly9373 committed:
Update Xiaohang PPP (#270)

* Update PPP and other documentation

* Update AboutUs

Kelly9373 committed:
Fix Xiaohang PPP (#278)

xantho09 committed:
Add diagram for Loan ID in Developer Guide (#274)

* Add first draft of add loan sequence diagram

* Add Sequence diagram for AddCommand in DG

* Update manual testing for Adding a Loan

* Fix checkstyle issues

* Rename AddLoanSequenceDiagram to LoanIdAddLoanSequenceDiagram

OrangeJuice7 committed:
Update UG/DG and PPP (#273)

* Update model class diagram

* Update model component section

* Update dev guide Bicycle Management section

* Remove Bicycle Management class diagram

* Add BicycleManagementAddLoanSequenceDiagram

* Add activity diagrams for BikeStatus

* Update Getting Started section in User Guide

* Rename LOAN_INDEX to LIST_INDEX in User Guide

* Fix nits in Commands section of User Guide

* Fix User Guide command specs

prokarius committed:
Fix PPP and increase content (#277)

* Upload David PPP

* Remove trailing whitespace

* Fix PPP

* Fix headers

* Extend PPP

* Remove accidentally pasted test cases

OrangeJuice7 committed:
Update AddCommand example (#276)

xantho09 committed:
Add Irham's PPP (#280)

Kelly9373 committed:
Fix some small errors in UG, LoanRate class and Email class. (#275)

* Fix some small errors in UG, LoanRate class and Email class.

* Fix regex in email, change the message in loanrate and update the UI of LoanCard.

* Fix regex error

wn committed:
Update weineng ppp (#279)

wn committed:
Version 1.4 release (#272)

wn committed:
Update PPP (#282)

wn committed:
Update UG intro (#281)
